src/Microsoft.Sbom.Api/Workflows/SbomConsolidationWorkflow.cs

```diff
@@ -3,41 +3,81 @@
     private IReadOnlyDictionary<string, ArtifactInfo> ArtifactInfoMap => configuration.ArtifactInfoMap.Value;
 #pragma warning restore IDE0051 // We'll use this soon.
 
-    public SbomConsolidationWorkflow(ILogger logger, IConfiguration configuration)
+    public SbomConsolidationWorkflow(ILogger logger, IConfiguration configuration, ISbomConfigFactory sbomConfigFactory, ISPDXFormatDetector sPDXFormatDetector, IFileSystemUtils fileSystemUtils, IMetadataBuilderFactory metadataBuilderFactory)
     {
         this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
         this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        this.sbomConfigFactory = sbomConfigFactory ?? throw new ArgumentNullException(nameof(sbomConfigFactory));
+        this.sPDXFormatDetector = sPDXFormatDetector ?? throw new ArgumentNullException(nameof(sPDXFormatDetector));
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.metadataBuilderFactory = metadataBuilderFactory ?? throw new ArgumentNullException(nameof(metadataBuilderFactory));
     }
 
     /// <inheritdoc/>
-#pragma warning disable CS1998 // Placeholder, will use async in the future.
     public virtual async Task<bool> RunAsync()
     {
         logger.Information("Placeholder SBOM consolidation workflow executed.");
 
-        return ValidateSourceSboms() && GeneratedConsolidatedSbom();
+        return await ValidateSourceSbomsAsync() && GeneratedConsolidatedSbom();
     }
-#pragma warning restore CS1998 // Placeholder, will use async in the future.
 
-    private bool ValidateSourceSboms()
+    private async Task<bool> ValidateSourceSbomsAsync()
     {
-        // TODO : Implement the source SBOMs.
-        return true;
+        var sbomsToValidate = ArtifactInfoMap.Select(artifact => GetSbomsToValidate(artifact.Key, artifact.Value))
+            .Where(l => l != null)
+            .SelectMany(l => l);
+        if (!sbomsToValidate.Any())
+        {
+            logger.Information($"No valid SBOMs detected.");
+            return false;
+        }
+
+        var validationWorkflows = sbomsToValidate
+            .Select(sbom => ValidateSourceSbomAsync(sbom.config, sbom.info));
+        var results = await Task.WhenAll(validationWorkflows);
+        return results.All(b => b);
+    }
+
+    private IEnumerable<(ISbomConfig config, ArtifactInfo info)> GetSbomsToValidate(string artifactPath, ArtifactInfo info)
+    {
+        var manifestDirPath = info?.ExternalManifestDir ?? fileSystemUtils.JoinPaths(artifactPath, Constants.ManifestFolder);
+        var isValidSpdxFormat = sPDXFormatDetector.TryGetSbomsWithVersion(manifestDirPath, out var detectedSboms);
+        if (!isValidSpdxFormat)
+        {
+            logger.Information($"No SBOMs located in {manifestDirPath} of a recognized SPDX format.");
+            return null;
```

Comment:
Given the scenarios in mind for consolidation, we might want to treat the "no SBOM was found" scenario the same as the "an invalid SBOM was found". The logging would be different, but the workflow would fail in both cases.

Reply:
With the current logic we log "No valid SBOMs detected." if there are no SBOMs found and return false in the validation step, causing the overall workflow to return false, so I think this case is already covered unless you prefer different behavior here.